Sporadic test_array_to_qimage_rgb failure using pyside2

The test sporadically fails both on Travis and during a manual test suite run ~after switching to unittest runner (#515)~ (edit: happens with both unittest and nose runners). I haven't managed to reproduce the error when running just that single test or the test module, seems to only happen when the whole test suite is run. 

For some reason the following pixels have different values than expected (`0xff4488cc`):
```
i=0, j=0: 0xff000000
i=1, j=0: 0xff000000
i=2, j=0: value differs between failures
i=4, j=0: 0xff000000
i=5, j=0: 0xff000000
i=6, j=0: value differs between failures
```

<details>
<summary>Non-informative failure traceback:</summary>

```
FAIL: test_array_to_qimage_rgb (pyface.ui.qt4.util.tests.test_image_helpers.TestImageHelpers)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/.edm/envs/pyface-test-3.6-pyside2/lib/python3.6/site-packages/pyface-7.1.0.dev10-py3.6.egg/pyface/ui/qt4/util/tests/test_image_helpers.py", line 79, in test_array_to_qimage_rgb
    for i in range(32) for j in range(64)
AssertionError: False is not true
```
</details>

--- a/pyface/ui/qt4/util/image_helpers.py
+++ b/pyface/ui/qt4/util/image_helpers.py
@@ -96,5 +96,6 @@ def array_to_QImage(array):
     elif channels == 4:
         image = QImage(data.data, width, height, bytes_per_line,
                        QImage.Format_ARGB32)
+    image._numpy_data = data
     return image
 

